pkg/agent/plugin/workloadattestor/docker/docker.go

```diff
@@ -47,12 +47,14 @@ type DockerPlugin struct {
 	cgroupContainerIndex int
 	fs                   cgroups.FileSystem
 	mtx                  *sync.RWMutex
```

Comment:
From another diff but a pointer to a `sync` primitive it a bit of a smell. You are returning a pointer from `New()` so the `sync.RWMutex` won't ever be copied.